lib/ethereumex/http_client.ex

```diff
@@ -17,7 +17,7 @@ defmodule Ethereumex.HttpClient do
 
   @spec post_request(binary()) :: {:ok | :error, any()}
   defp post_request(payload) do
-    response = rpc_url() |> HTTPoison.post(payload, [{"Content-Type", "application/json"}])
+    response = rpc_url() |> HTTPoison.post(payload, [{"Content-Type", "application/json"}], Ethereumex.Config.http_options())
```

Comment:
I think this is a good idea. We should implement this to generic Client requests as well, useful for things like timeout threshold of HTTPoision. What do you think @ayrat555 ?

Reply:
Yeas, it's nice feature. I am just waiting for @hayesgm to move to URL option in config. And it's good to merge.